pkg/scheduler/algorithmprovider/registry_test.go

```diff
@@ -17,83 +17,163 @@ limitations under the License.
+			PreFilter: &schedulerapi.PluginSet{
+				Enabled: []schedulerapi.Plugin{
+					{Name: noderesources.FitName},
+					{Name: nodeports.Name},
+					{Name: interpodaffinity.Name},
+					{Name: podtopologyspread.Name},
+				},
+			},
+			Filter: &schedulerapi.PluginSet{
+				Enabled: []schedulerapi.Plugin{
+					{Name: nodeunschedulable.Name},
+					{Name: noderesources.FitName},
+					{Name: nodename.Name},
+					{Name: nodeports.Name},
+					{Name: nodeaffinity.Name},
+					{Name: volumerestrictions.Name},
+					{Name: tainttoleration.Name},
+					{Name: nodevolumelimits.EBSName},
+					{Name: nodevolumelimits.GCEPDName},
+					{Name: nodevolumelimits.CSIName},
+					{Name: nodevolumelimits.AzureDiskName},
+					{Name: volumebinding.Name},
+					{Name: volumezone.Name},
+					{Name: interpodaffinity.Name},
+					{Name: podtopologyspread.Name},
+				},
+			},
+			PostFilter: &schedulerapi.PluginSet{
+				Enabled: []schedulerapi.Plugin{
+					{Name: interpodaffinity.Name},
+					{Name: tainttoleration.Name},
+					{Name: podtopologyspread.Name},
+				},
+			},
+			Score: &schedulerapi.PluginSet{
+				Enabled: []schedulerapi.Plugin{
+					{Name: noderesources.BalancedAllocationName, Weight: 1},
+					{Name: imagelocality.Name, Weight: 1},
+					{Name: interpodaffinity.Name, Weight: 1},
+					{Name: noderesources.LeastAllocatedName, Weight: 1},
+					{Name: nodeaffinity.Name, Weight: 1},
+					{Name: nodepreferavoidpods.Name, Weight: 10000},
+					{Name: defaultpodtopologyspread.Name, Weight: 1},
+					{Name: tainttoleration.Name, Weight: 1},
+					{Name: podtopologyspread.Name, Weight: 1},
+					{Name: noderesources.ResourceLimitsName, Weight: 1},
+				},
+			},
+		},
+		FrameworkPluginConfig: []schedulerapi.PluginConfig{
+			{
+				Name: interpodaffinity.Name,
+				Args: runtime.Unknown{Raw: []byte(fmt.Sprintf(`{"hardPodAffinityWeight":%d}`, hardPodAffinityWeight))},
+			},
+		},
+	}
 
-			volumeBinder := volumebinder.NewVolumeBinder(
-				client,
-				informerFactory.Core().V1().Nodes(),
-				informerFactory.Storage().V1().CSINodes(),
-				informerFactory.Core().V1().PersistentVolumeClaims(),
-				informerFactory.Core().V1().PersistentVolumes(),
-				informerFactory.Storage().V1().StorageClasses(),
-				time.Second,
-			)
-			providerRegistry := NewRegistry(1)
-			config := providerRegistry[tc.provider]
-			fwk, err := framework.NewFramework(
-				plugins.NewInTreeRegistry(&plugins.RegistryArgs{
-					VolumeBinder: volumeBinder,
-				}),
-				config.FrameworkPlugins,
-				config.FrameworkPluginConfig,
-				framework.WithClientSet(client),
-				framework.WithInformerFactory(informerFactory),
-				framework.WithSnapshotSharedLister(nodeinfosnapshot.NewEmptySnapshot()),
-			)
-			if err != nil {
-				t.Fatalf("error initializing the scheduling framework: %v", err)
-			}
-			wantPlugins := fwk.ListPlugins()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ResourceLimitsPriorityFunction, true)()
```

Comment:
Can we add a case to ensure PodTopologySpread plugins are not enabled when the featuregate is disabled?

Reply:
done.